smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpBindingProtocolGenerator.kt

```diff
@@ -274,7 +274,9 @@ abstract class HttpBindingProtocolGenerator : ProtocolGenerator {
         }
 
         resolver.determineRequestContentType(op)?.let { contentType ->
-            writer.write("builder.headers.setMissing(\"Content-Type\", #S)", contentType)
+            writer.openBlock("if (builder.body !is HttpBody.Empty) {", "}") {
+                writer.write("builder.headers.setMissing(\"Content-Type\", #S)", contentType)
+            }
```

Comment:
**Style**: `openBlock` is part of the Java **smithy** interface and it accomplishes the desired task here but we often prefer [our own Kotlin extension methods](https://github.com/awslabs/smithy-kotlin/blob/main/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/CodeWriterExt.kt) such as `withBlock`, which includes the writer as a lambda receiver. This would allow you to slightly simplify this code by omitting the explicit writer on L278:

```kotlin
writer.withBlock("if (builder.body !is HttpBody.Empty) {", "}") {
    write("builder.headers.setMissing(\"Content-Type\", #S)", contentType)
}
```

Reply:
I included `Content-Length` verification in the new protocol tests, the outcomes for both with and without a body seem to be correct there.

Reply:
FYI `Content-Length` is a bit more nuanced anyway since each HTTP engine supports how it is set differently. For instance the ktor based engine doesn't allow content-length to be set and instead will set it based on the `HttpBody` length property.